obsidian.el

```diff
@@ -353,21 +357,29 @@ Optional argument ARG word to complete."
   (obsidian-update-tags-list)
   (obsidian--update-all-from-front-matter))
 
-(defun obsidian--request-link ()
-  "Service function to request user for link iput."
+(defun obsidian--request-link (&optional toggle-path)
+  "Service function to request user for link input.
+
+TOGGLE-PATH is a boolean that will toggle the behavior of
+`obsidian-use-vault-path' for this single link insertion."
   (let* ((all-files (->> (obsidian-list-all-files) (-map (lambda (f) (file-relative-name f obsidian-directory)))))
          (region (when (use-region-p)
                    (buffer-substring-no-properties (region-beginning) (region-end))))
          (chosen-file (completing-read "Link: " all-files))
-         (default-description (-> chosen-file file-name-nondirectory file-name-sans-extension))
-         (description (read-from-minibuffer "Description (optional): " (or region default-description))))
-    (list :file chosen-file :description description)))
+         (filename-only (file-name-nondirectory chosen-file))
+         (default-description (file-name-sans-extension filename-only))
+         (description (read-from-minibuffer "Description (optional): " (or region default-description)))
+         (file-link (if obsidian-links-use-vault-path
+                        (if toggle-path filename-only chosen-file)
+                      (if toggle-path chosen-file filename-only)))
+         )
+    (list :file file-link :description description)))
 
 ;;;###autoload
 (defun obsidian-insert-wikilink ()
   "Insert a link to file in wikiling format."
   (interactive)
-  (let* ((file (obsidian--request-link))
+  (let* ((file (obsidian--request-link current-prefix-arg))

```

Comment:
In my opinion using `(interactive "P")` more clearly communicates the intention at first glance. Is there a reason to use the `current-prefix-arg` instead? Never seen this one before :)